pkg/client/restclient/request_test.go

```diff
@@ -838,6 +838,36 @@ func TestBackoffLifecycle(t *testing.T) {
 	}
 }
 
+type testBackoffManager struct {
+	t              *testing.T
+	expectedSleeps []time.Duration
+}
+
+func (b *testBackoffManager) UpdateBackoff(actualUrl *url.URL, err error, responseCode int) {
+}
+
+func (b *testBackoffManager) CalculateBackoff(actualUrl *url.URL) time.Duration {
+	return time.Duration(0)
+}
+
+func (b *testBackoffManager) Sleep(d time.Duration) {
+	if len(b.expectedSleeps) == 0 {
+		b.t.Errorf("unexpected sleep: %v", d)
+	} else {
+		expected := b.expectedSleeps[0]
+		b.expectedSleeps = b.expectedSleeps[1:]
+		if expected != d {
+			b.t.Errorf("incorrect sleep: expected %v, got %v", expected, d)
+		}
+	}
+}
+
+func (b *testBackoffManager) Verify() {
+	if len(b.expectedSleeps) != 0 {
+		b.t.Errorf("some sleeps weren't called: %v", b.expectedSleeps)
+	}
+}
```

Comment:
`testBackoffManager` should probably just record the sleep calls and leave the verification to the caller. This would allow for more test specific verification, put the error messages in the test cases, and de-tangle this struct from a `*testing.T` member.